platform/app/services/temporal/activities/process_page_activity.rb

```
@@ -0,0 +1,88 @@
+      end
+
+      def create_sync_read_record(records)
+        SyncReadRecord.create!(
+          sync_run: @sync_run,
+          sync: @sync,
+          data: records
+        )
+      rescue ActiveRecord::RecordInvalid => e
+        raise unless e.record.errors[:signature]&.include?("has already been taken")
+
+          activity.logger.info("Skipping duplicate page for sync_id: #{@sync.id}")
```

rubocop (1.68.0)

[convention] Inconsistent indentation detected.

(Layout/IndentationConsistency)